modules/develop/pages/produce-data/configure-producers.adoc

```diff
@@ -232,6 +232,24 @@ verbose format, which impacts your compression speeds and network throughput.
 Alternatively, going with AVRO or Protobuf allows you to only define the schema
 in one place, while also enabling features like schema evolution.
 
+[[broker_timestamps]]
+=== Broker timestamps
+
+Redpanda employs a unique strategy to help ensure accuracy in retention operations. Every time a producer posts a message to a topic, Redpanda records the broker's system date and time in the message's `broker_timestamp` property. This helps ensure retention functions as intended in cases where a producer submits a message with a created timestamp skewed from the broker's time. Closed segments are only eligible for deletion under Redpanda's retention policy when the age of all messages in the segment exceed a set threshold. Using broker timestamps ensures that skewed messages are cleaned up appropriately where they may otherwise appear to never exceed the retention policy threshold.
```

Comment:
```suggestion
Redpanda employs a unique strategy to help ensure the accuracy of retention operations. In this strategy, closed segments are only eligible for deletion when the age of all messages in the segment exceeds a <<set-time-based-retention, configured threshold>>. However, when a producer sends a message to a topic, the timestamp set by the producer may not accurately reflect the time the message reaches the broker. To address this time skew, each time a producer sends a message to a topic, Redpanda records the broker's system date and time in the `broker_timestamp` property of the message. This property helps maintain accurate retention policies, even when the message's creation timestamp deviates from the broker's time.
```

Reply:
I wonder if this flow is correct. Shouldn't we introduce the idea of setting time-based retention first before we introduce broker timestamps?

Reply:
I like the re-ordering here. This paragraph got mangled by committee and this reads a lot better.